db/silkworm/etl/file_provider.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+   Copyright 2020 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "file_provider.hpp"
+
+namespace silkworm::etl{
+
+FileProvider::FileProvider(int id) {
+    filename_ = "tmp" + std::to_string(id);
+    file_.open(filename_, std::ios_base::in | std::ios_base::out | std::ios_base::trunc);
```

Comment:
There is no check whether or not the file is good.
Do we have space on disk ? The opening of the file was succesfull ? (Maybe we don't have the right to write in this directory=